src/Microsoft.Health.Dicom.Blob/Features/Export/AzureBlobExportSinkProvider.cs

```diff
@@ -125,6 +134,33 @@ protected override async Task<AzureBlobExportOptions> SecureSensitiveInfoAsync(A
     protected override Task ValidateAsync(AzureBlobExportOptions options, CancellationToken cancellationToken = default)
     {
         List<ValidationResult> results = options.Validate(new ValidationContext(this)).ToList();
+
+        // Run additional validation based on the sink provider options if the options themselves are configured correctly.
+        // We need to ensure that they do not contain SAS Tokens or use public access if it is disallowed
+        if (results.Count == 0)
+        {
+            if (options.BlobContainerUri != null)
+            {
+                if (options.BlobContainerUri.Query.Length > 1)
```

Comment:
Is the assumption here that the any query string is a sas token? Does it have a specific query key?